pkg/env/v1alpha1/env.go

```diff
@@ -18,43 +18,141 @@ package v1alpha1
+	// to delete jiva cas volumes
+	CASTemplateToDeleteJivaVolumeENVK ENVKey = "OPENEBS_IO_JIVA_CAS_TEMPLATE_TO_DELETE_VOLUME"
+
+	// CASTemplateToCreateCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to create cstor cas volumes
+	CASTemplateToCreateCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_CREATE_VOLUME"
+
+	// CASTemplateToReadCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to read cstor cas volumes
+	CASTemplateToReadCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_READ_VOLUME"
+
+	// CASTemplateToDeleteCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to delete cstor cas volumes
+	CASTemplateToDeleteCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_DELETE_VOLUME"
+
+	// CASTemplateToCreatePoolENVK is the ENV key that specifies the CAS Template
+	// to create storage pool
+	CASTemplateToCreatePoolENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_TO_CREATE_POOL"
+
+	// CASTemplateToDeletePoolENVK is the ENV key that specifies the CAS Template
+	// to delete storage pool
+	CASTemplateToDeletePoolENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_TO_DELETE_POOL"
 )
 
+// EnvironmentSetter abstracts setting of environment variable
+type EnvironmentSetter func(envKey ENVKey, value string) (err error)
+
 // EnvironmentGetter abstracts fetching value from an environment variable
-type EnvironmentGetter func(envKey string) (value string)
+type EnvironmentGetter func(envKey ENVKey) (value string)
+
+// EnvironmentLookup abstracts looking up an environment variable
+type EnvironmentLookup func(envKey ENVKey) (value string, present bool)
+
+// Set sets the provided environment variable
+//
+// NOTE:
+//  This is an implementation of EnvironmentSetter
+func Set(envKey ENVKey, value string) (err error) {
+	return os.Setenv(string(envKey), value)
+}
 
 // Get fetches value from the provided environment variable
 //
 // NOTE:
 //  This is an implementation of EnvironmentGetter
-func Get(envKey string) (value string) {
-	return getEnv(envKey)
+func Get(envKey ENVKey) (value string) {
+	return getEnv(string(envKey))
 }
 
-// EnvironmentLookup abstracts looking up an environment variable
-type EnvironmentLookup func(envKey string) (value string, present bool)
-
 // Lookup looks up an environment variable
 //
 // NOTE:
 //  This is an implementation of EnvironmentLookup
-func Lookup(envKey string) (value string, present bool) {
-	return lookupEnv(envKey)
+func Lookup(envKey ENVKey) (value string, present bool) {
+	return lookupEnv(string(envKey))
+}
+
+// Truthy returns boolean based on the environment variable's value
+//
+//  The value can be truthy or falsy
+func Truthy(envKey ENVKey) (truth bool) {
+	v, found := Lookup(envKey)
+	if !found {
+		return
+	}
+	truth, _ = strconv.ParseBool(v)
+	return
+}
+
+// LookupS looks up an environment variable and returns false if environment
+// variable is not present
+func LookupS(envKey ENVKey) string {
```

Comment:
Some better name required. Suffix 'S' not making any sense.

Reply:
Will do something about it @princerachit